File: Programmers/Hashing/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string, int>names;
    string answer = "";
    for (auto str : completion) {
        names[str]++;
    }
    for (auto str : participant) {
        if (names[str]) {
            names[str]--;
        }
        else {
            answer = str;
        }
    }
    return answer;
}

File: Programmers/Hashing/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    s = dict()
    for str in completion:
        if str in s:
            s[str]+=1
        else:
            s[str]=1
    for str in participant:
        if str in s:
            s[str]-=1
            if s[str]==0:
                del s[str]
        else:
            return str